pkg/apis/core/v1alpha1/cmd_types.go

```diff
@@ -93,7 +93,14 @@ type CmdSpec struct {
 	// Restarts can happen even if the command is already done.
 	//
 	// Logs of the current process after the restart are discarded.
-	RestartOn *RestartOnSpec `json:"restartOn,omitempty" protobuf:"bytes,5,opt,name=restartOn"`
+	RestartOn *TriggerSpec `json:"restartOn,omitempty" protobuf:"bytes,5,opt,name=restartOn"`
+
+	// Indicates objects that can trigger a run of this command.
```

Comment:
We will probably need to support the case where:

1) User creates a Button and CmdA
2) User clicks the Button a few times
3) User deletes CmdA and creates CmdB

where we wouldn't want CmdB to run until the next button press. How would we model that scenario?

Reply:
I've added a StartAfter to StartOn